samcli/local/docker/lambda_build_container.py

```diff
@@ -83,16 +88,25 @@ def __init__(  # pylint: disable=too-many-locals
         )
 
         if image is None:
-            image = LambdaBuildContainer._get_image(runtime, architecture)
+            # use specified_workflow to get image if exists, otherwise use runtime
+            runtime_to_get_image = specified_workflow if specified_workflow else runtime
+            image = LambdaBuildContainer._get_image(runtime_to_get_image, architecture)
         entry = LambdaBuildContainer._get_entrypoint(request_json)
-        cmd = []
+        cmd: List[str] = []
 
+        mount_mode = "rw" if mount_with_write else "ro"
         additional_volumes = {
             # Manifest is mounted separately in order to support the case where manifest
             # is outside of source directory
-            manifest_dir: {"bind": container_dirs["manifest_dir"], "mode": "ro"}
+            manifest_dir: {"bind": container_dirs["manifest_dir"], "mode": mount_mode}
         }
 
+        host_tmp_dir = None
+        if mount_with_write:
+            # Mounting tmp dir on the host as ``/tmp/samcli`` on container, which gives current user write permissions
+            host_tmp_dir = os.path.join(source_dir, str(uuid4()))
```

Comment:
Let's use utility functions to create tmp directory rather than creating one in the source folder. See example utility here: https://github.com/aws/aws-sam-cli/blob/3a7a4c39d62456b623d4183857c6e253e5236cef/samcli/lib/package/local_files_utils.py#L14

Reply:
Instead of `source_dir`, now using `../.aws-sam/build/`